File: src/tests/fidl/source_compatibility/protocol-method-add/llcpp/before.cc
// Copyright 2020 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include <fidl/test/addmethod/llcpp/fidl.h>  // nogncheck
namespace fidl_test = llcpp::fidl::test::addmethod;

class Impl final : public fidl_test::ExampleProtocol::Interface {
 public:
  void ExistingMethod(ExistingMethodCompleter::Sync& completer) final {}
};

int main(int argc, const char** argv) {
  Impl server{};
  return 0;
}

File: src/tests/fidl/source_compatibility/protocol-method-add/go/before.go
// Copyright 2020 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package main

import (
	lib "fidl/fidl/test/addmethod"
	"syscall/zx/fidl"
)

type impl struct {
	lib.ExampleProtocolWithCtxTransitionalBase
}

func (*impl) ExistingMethod(fidl.Context) error {
	return nil
}

// Assert that the impl implements the ExampleProtocol interface
var _ lib.ExampleProtocolWithCtx = &impl{}

func main() {}
